bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Device.java

```diff
@@ -55,6 +58,7 @@ public abstract class Device implements Drawable {
 	String[] loadedFonts;
 
 	volatile boolean disposed;
+	private Set<Resource> resourcesWithZoomSupport  = Collections.synchronizedSet(new HashSet<>());
```

Comment:
See general comment about `synchronized`, in any case 

```suggestion
	private Set<Resource> resourcesWithZoomSupport  = ConcurrentHashMap.newKeySet();
```

might be a better alternative these days.